Desktop Enrollment integration test fails on Firefox 152 (about:config flip_pref no longer toggles)

## Problem

The `update_firefox_desktop_release` PRs fail `Desktop Enrollment (1)` once the release bump crosses Firefox 151 → 152, with a misleading `selenium ... TimeoutException: Summary Page: Unable to find live status`. Re-running does not help. This **blocks every desktop release-version bump from merging** (e.g. #16021, earlier #15983).

It is not flaky. The Desktop Enrollment workflow is hardcoded to `FIREFOX_CHANNEL: release`, so every run installs the release Firefox from `experimenter/tests/firefox_desktop_release_build.env`. Clean A/B: release 151.0.4 passes everywhere; release 152.0.x fails only on the PRs that change that file. Same commit, same test, only the browser binary differs.

## Real failure

The CI text log shows only the rerun's masked `TimeoutException`. The uploaded test-report artifact has the actual first-attempt traceback:

```
about_config.flip_pref(pref_name)
> assert about_config.get_pref_value(pref_name) == "false"
E AssertionError: assert 'true' == 'false'
test_desktop_client_integrations.py:217
```

Enrollment fires and the sticky experiment correctly sets `sticky.targeting.test.pref` to `true` (so Nimbus enrollment and sticky setPref both work on 152). `flip_pref()` is meant to toggle it back to `false`, but the toggle is a no-op, so the assert fails. The "Unable to find live status" timeout appears only on the rerun, because the failed first attempt leaves a half-launched experiment in the single-worker stack and the fresh rerun experiment can't sync to Live.

## Root cause

`flip_pref` in `experimenter/tests/integration/nimbus/pages/browser/__init__.py` single-clicks the cell:

```python
el.find_element(By.CSS_SELECTOR, ".cell-edit").click()   # clicks <td class="cell-edit">
```

Firefox 152's about:config click handler now ignores single-clicks that aren't on a button — `if (event.target.localName != "button") { return; }`. In 151 a single click on the cell toggled the boolean; in 152 you must click the inner `<button>` (or double-click). The test single-clicks the `<td>`, so the bool is never toggled.

This is a test-maintenance issue, not a Nimbus/product regression.

## Proposed fix

Click the toggle button inside the edit cell instead of the cell:

```python
el.find_element(By.CSS_SELECTOR, ".cell-edit button").click()
```

`flip_pref` is the shared pref-toggle helper, so this protects any other test that flips a pref.

## Acceptance criteria

- `test_check_telemetry_sticky_targeting[FIREFOX_DESKTOP]` passes against Firefox release 152.x.
- The desktop release-version bump PR (#16021) goes green once rebased on this fix.

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/EXP-7036)

--- a/experimenter/tests/integration/nimbus/pages/browser/__init__.py
+++ b/experimenter/tests/integration/nimbus/pages/browser/__init__.py
@@ -3,8 +3,6 @@
 import time
 
 from pypom import Page
-from selenium.webdriver.common.by import By
-from selenium.webdriver.support import expected_conditions as EC
 
 
 class Browser:
@@ -19,16 +17,8 @@ class Browser:
 class AboutConfig(Page):
     URL_TEMPLATE = "about:config"
 
-    _search_bar_locator = (By.ID, "about-config-search")
-    _row_locator = (By.CSS_SELECTOR, "tr > td > span > span")
-    _pref_locator = (By.CSS_SELECTOR, "tr > th > span")
-
     def __init__(self, selenium, **kwargs):
         super().__init__(selenium, timeout=80, **kwargs)
-
-    def wait_for_page_to_load(self):
-        self.wait.until(EC.presence_of_element_located(self._search_bar_locator))
-        return self
 
     def wait_for_pref_flip(self, pref, pref_value, action=None, pref_type=str):
         timeout = time.time() + 60 * 5
@@ -61,44 +51,35 @@ class AboutConfig(Page):
                     return
         raise (error)
 
-    def flip_pref(self, pref):
-        timeout = time.time() + 15
-        error = None
-        while time.time() < timeout:
-            try:
-                search_bar = self.find_element(*self._search_bar_locator)
-                search_bar.send_keys(pref)
-                self.wait.until(EC.presence_of_element_located(self._row_locator))
-                actual_row = (By.CSS_SELECTOR, "table > tr")
-                new_pref_locator = (By.CSS_SELECTOR, "th > span")
-                elements = self.find_elements(*actual_row)
-                el = next(
-                    element
-                    for element in elements
-                    if pref in element.find_element(*new_pref_locator).text
-                )
-                el.find_element(By.CSS_SELECTOR, ".cell-edit").click()
-            except Exception as e:
-                error = e
-                time.sleep(2)
-                self.selenium.get("about:config")
-                self.wait_for_page_to_load()
-            else:
-                return True
-        raise (error)
+    def get_pref_value(self, pref, pref_type=str):
+        match pref_type:
+            case _ if pref_type is bool:
+                method = "getBoolPref"
+            case _ if pref_type is int:
+                method = "getIntPref"
+            case _ if pref_type is str:
+                method = "getStringPref"
+            case _:
+                raise TypeError(f"Unsupported pref type {pref_type}")
+        with self.selenium.context(self.selenium.CONTEXT_CHROME):
+            return self.selenium.execute_script(
+                f"return Services.prefs.{method}(arguments[0]);", pref
+            )
 
-    def get_pref_value(self, pref):
-        self.selenium.get("about:config")
-        self = self.wait_for_page_to_load()
-        search_bar = self.find_element(*self._search_bar_locator)
-        search_bar.send_keys(pref)
-        self.wait.until(EC.presence_of_element_located(self._row_locator))
-        actual_row = (By.CSS_SELECTOR, "table > tr")
-        new_pref_locator = (By.CSS_SELECTOR, "th > span")
-        elements = self.find_elements(*actual_row)
-        el = next(
-            element
-            for element in elements
-            if pref in element.find_element(*new_pref_locator).text
-        )
-        return el.find_element(By.CSS_SELECTOR, ".cell-value").text
+    def set_pref(self, pref, value):
+        match value:
+            case bool():
+                method = "setBoolPref"
+            case int():
+                method = "setIntPref"
+            case str():
+                method = "setStringPref"
+            case _:
+                raise TypeError(f"Unsupported pref type {type(value)}")
+        with self.selenium.context(self.selenium.CONTEXT_CHROME):
+            self.selenium.execute_script(
+                f"Services.prefs.{method}(arguments[0], arguments[1]);", pref, value
+            )
+
+    def flip_pref(self, pref):
+        self.set_pref(pref, not self.get_pref_value(pref, pref_type=bool))

--- a/experimenter/tests/integration/nimbus/test_desktop_client_integrations.py
+++ b/experimenter/tests/integration/nimbus/test_desktop_client_integrations.py
@@ -133,7 +133,6 @@ def test_check_telemetry_pref_flip(
             "is_rollout": True,
         },
     )
-    about_config = about_config.open().wait_for_page_to_load()
     about_config.wait_for_pref_flip(
         "nimbus.qa.pref-1", "default", action=trigger_experiment_loader
     )
@@ -151,7 +150,6 @@ def test_check_telemetry_pref_flip(
         if time.time() > timeout:
             raise AssertionError("Experiment enrollment was never seen in ping Data")
 
-    about_config = about_config.open().wait_for_page_to_load()
     about_config.wait_for_pref_flip(
         "nimbus.qa.pref-1", "test_string_automation", action=trigger_experiment_loader
     )
@@ -168,7 +166,6 @@ def test_check_telemetry_pref_flip(
         control = telemetry_event_check(experiment_slug, "unenrollment")
         if time.time() > timeout:
             raise AssertionError("Experiment unenrollment was never seen in ping Data")
-    about_config = about_config.open().wait_for_page_to_load()
     about_config.wait_for_pref_flip(
         "nimbus.qa.pref-1", "default", action=trigger_experiment_loader
     )
@@ -209,12 +206,11 @@ def test_check_telemetry_sticky_targeting(
             raise AssertionError("Experiment enrollment was never seen in ping Data")
 
     # flip pref
-    about_config = about_config.open().wait_for_page_to_load()
     about_config.wait_for_pref_flip(
         pref_name, True, action=trigger_experiment_loader, pref_type=bool
     )
     about_config.flip_pref(pref_name)
-    assert about_config.get_pref_value(pref_name) == "false"
+    assert about_config.get_pref_value(pref_name, pref_type=bool) is False
     # check experiment doesn't unenroll after pref flip
     control = False
     timeout = time.time() + 60
@@ -238,4 +234,4 @@ def test_check_telemetry_sticky_targeting(
         if time.time() > timeout:
             raise AssertionError("Experiment unenrollment was never seen in ping Data")
     # check pref still matches user change
-    assert about_config.get_pref_value(pref_name) == "false"
+    assert about_config.get_pref_value(pref_name, pref_type=bool) is False

